### PR TITLE
🚑 Fix nft storage upload

### PR DESCRIPTION
**Pull request by bartektrybala**
## Description

- nft_storage `upload` replaced with `filebase_upload` and `pin_file_to_ipfs`
- new field `nft_storage` in Tree model
- upload file process adjusted to use the new storage

## dev changes
- new libraries: `pydantic`, `responses`
- stubs for `boto3` and `protobuf`
- `live_request` pytest marker for live testing

### CODE REVIEW

**Comment by bartektrybala** [comment]
@tulustul
How to update the secrets with new values for `NFT_STORAGE_API_KEY` and `NFT_STORAGE_SECRET_ACCESS_KEY`?How to test it on the dev environment?

File: backend/replant/integrations/nft_storage.py
Comment (tulustul) [inline backend/replant/integrations/nft_storage.py:1]: Can we change the name of the file to `filebase.py`?

File: backend/replant/tests/integrations/test_nft_storage.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    mock_boto_client: mock.MagicMock,
+    image_to_upload: io.BytesIO,
+) -> None:
+    # mocks
+    mock_s3 = mock.MagicMock()
+    mock_boto_client.return_value = mock_s3
+    mock_s3.put_object.return_value = SUCCESS_FILEBASE_UPLOAD_IMAGE_RESPONSE
+
+    # when
+    response = nft_storage.filebase_upload(
```
Comment (tulustul) [inline backend/replant/tests/integrations/test_nft_storage.py:50]: maybe `filebase.upload`

File: backend/replant/integrations/nft_storage.py [hunk 2]
```diff
@@ -1,44 +1,80 @@
...
+
+@retry(delay=1, backoff=2, tries=3)
+def filebase_upload(dto: FileDto) -> UploadResponse:
+    s3 = boto3.client(
+        service_name="s3",
+        endpoint_url=env.NFT_STORAGE_API_URL,
+        aws_access_key_id=env.NFT_STORAGE_ACCESS_KEY,
+        aws_secret_access_key=env.NFT_STORAGE_SECRET_ACCESS_KEY,
+    )
+    upload_resposne = s3.put_object(
```
Comment (tulustul) [inline backend/replant/integrations/nft_storage.py:40]: typo `resposne` -> `response`

File: backend/replant/models/tree.py [hunk 3]
```diff
@@ -46,20 +48,29 @@ class MintingState(models.TextChoices):
...
     )
+
     image = models.ImageField(upload_to=image_upload_to)
     image_cid = models.URLField(
         max_length=128, default="", blank=True, verbose_name="image CID"
     )
     metadata_cid = models.URLField(
         max_length=128, default="", blank=True, verbose_name="metadata CID"
     )
+    nft_storage = models.CharField(
```
Comment (tulustul) [inline backend/replant/models/tree.py:70]: `nft_storage` is misleading because it's the name of a service we were using.
`storage_provider` maybe?

**Review by tulustul — APPROVED** [review]

**Review by tulustul — APPROVED** [review]

**Pull request merged**